conan/tools/premake/premakedeps.py

```diff
@@ -37,74 +109,200 @@ def __init__(self, deps_cpp_info):
 
 
 class PremakeDeps(object):
+    """
+    PremakeDeps class generator
+    conandeps.premake5.lua: unconditional import of all *direct* dependencies only
+    """
 
     def __init__(self, conanfile):
+        """
+        :param conanfile: ``< ConanFile object >`` The current recipe object. Always use ``self``.
+        """
+
         self._conanfile = conanfile
 
+        # Tab configuration
+        self.tab = "\t"
+
+        # Return value buffer
+        self.output_files = {}
+        # Extract configuration and architecture form conanfile
+        self.configuration = conanfile.settings.build_type
+        self.architecture = conanfile.settings.arch
+
     def generate(self):
+        """
+        Generates ``conan_<pkg>_vars_<config>.premake5.lua``, ``conan_<pkg>_<config>.premake5.lua``,
+        and ``conan_<pkg>.premake5.lua`` files into the ``conanfile.generators_folder``.
+        """
+
         check_duplicated_generator(self, self._conanfile)
         # Current directory is the generators_folder
         generator_files = self.content
         for generator_file, content in generator_files.items():
             save(generator_file, content)
 
-    def _get_cpp_info(self):
-        ret = CppInfo()
-        for dep in self._conanfile.dependencies.host.values():
-            dep_cppinfo = dep.cpp_info.copy()
-            dep_cppinfo.set_relative_base_folder(dep.package_folder)
-            # In case we have components, aggregate them, we do not support isolated "targets"
-            dep_cppinfo.aggregate_components()
-            ret.merge(dep_cppinfo)
-        return ret
+    def _config_suffix(self):
+        props = [("Configuration", self.configuration),
+                 ("Platform", self.architecture)]
+        name = "".join("_%s" % v for _, v in props)
+        return name.lower()
+    
+    def _output_lua_file(self, filename, content):
+        self.output_files[filename] = "\n".join(["#!lua", *content])
+
+    def _indent_string(self, string, indent=1):
+        return "\n".join([
+            f"{self.tab * indent}{line}" for line in list(filter(None, string.splitlines()))
+        ])
+
+    def _premake_filtered(self, content, configuration, architecture, indent=0):
```

Comment:
These filters are quite unknown for others (like us) not knowing Premake. A short explanation or a link to somewhere that defines what are those would be nice.